VoodooPS2: error: The one definition rule is violated, different classes/structs have the same name 'virtual_finger_state'

```
VoodooPS2Trackpad/VoodooPS2Elan.h:22:1: error: The one definition rule is violated, different classes/structs have the same name 'virtual_finger_state' [ctuOneDefinitionRuleViolation]
struct virtual_finger_state {
^
VoodooPS2Trackpad/VoodooPS2SynapticsTouchPad.h:173:1: note: The one definition rule is violated, different classes/structs have the same name 'virtual_finger_state'
struct virtual_finger_state {
^
VoodooPS2Trackpad/VoodooPS2Elan.h:22:1: note: The one definition rule is violated, different classes/structs have the same name 'virtual_finger_state'
struct virtual_finger_state {
^
```

--- a/VoodooPS2Trackpad/VoodooPS2ALPSGlidePoint.cpp
+++ b/VoodooPS2Trackpad/VoodooPS2ALPSGlidePoint.cpp
@@ -3406,7 +3406,7 @@ void ApplePS2ALPSGlidePoint::assignVirtualFinger(int physicalFinger) {
         return;
     }
     for (int j = 0; j < MAX_TOUCHES; j++) {
-        virtual_finger_state &vfj = virtualFingerStates[j];
+        synaptics_virtual_finger_state &vfj = virtualFingerStates[j];
         if (!vfj.touch) {
             fingerStates[physicalFinger].virtualFingerIndex = j;
             vfj.touch = true;
@@ -3418,7 +3418,7 @@ void ApplePS2ALPSGlidePoint::assignVirtualFinger(int physicalFinger) {
     }
 }
 
-void ApplePS2ALPSGlidePoint::assignFingerType(virtual_finger_state &vf) {
+void ApplePS2ALPSGlidePoint::assignFingerType(synaptics_virtual_finger_state &vf) {
     vf.fingerType = kMT2FingerTypeUndefined;
     for (MT2FingerType i = kMT2FingerTypeIndexFinger; i < kMT2FingerTypeCount; i = (MT2FingerType)(i + 1))
         if (freeFingerTypes[i]) {
@@ -3780,7 +3780,7 @@ bool ApplePS2ALPSGlidePoint::renumberFingers() {
             IOLog("alps_parse_hw_state: ERROR: invalid physical finger %d\n", fi.virtualFingerIndex);
             continue;
         }
-        virtual_finger_state &fiv = virtualFingerStates[fi.virtualFingerIndex];
+        synaptics_virtual_finger_state &fiv = virtualFingerStates[fi.virtualFingerIndex];
         fiv.x_avg.filter(fi.x);
         fiv.y_avg.filter(fi.y);
         fiv.pressure = fi.z;

--- a/VoodooPS2Trackpad/VoodooPS2ALPSGlidePoint.h
+++ b/VoodooPS2Trackpad/VoodooPS2ALPSGlidePoint.h
@@ -182,7 +182,7 @@ struct alps_hw_state {
     int virtualFingerIndex;
 };
 
-struct virtual_finger_state {
+struct synaptics_virtual_finger_state {
     SimpleAverage<int, 5> x_avg;
     SimpleAverage<int, 5> y_avg;
     uint8_t pressure;
@@ -526,7 +526,7 @@ private:
     uint32_t physical_max_y {0};
 
     alps_hw_state fingerStates[MAX_TOUCHES] {};
-    virtual_finger_state virtualFingerStates[MAX_TOUCHES] {};
+    synaptics_virtual_finger_state virtualFingerStates[MAX_TOUCHES] {};
     bool freeFingerTypes[kMT2FingerTypeCount];
 
     static_assert(MAX_TOUCHES <= kMT2FingerTypeLittleFinger, "Too many fingers for one hand");
@@ -661,7 +661,7 @@ private:
 
     int dist(int physicalFinger, int virtualFinger);
     void assignVirtualFinger(int physicalFinger);
-    void assignFingerType(virtual_finger_state &vf);
+    void assignFingerType(synaptics_virtual_finger_state &vf);
     void freeAndMarkVirtualFingers();
     int upperFingerIndex() const;
     const alps_hw_state& upperFinger() const;

--- a/VoodooPS2Trackpad/VoodooPS2Elan.h
+++ b/VoodooPS2Trackpad/VoodooPS2Elan.h
@@ -19,7 +19,7 @@
 #include "VoodooInputMultitouch/VoodooInputEvent.h"
 #include "VoodooPS2TrackpadCommon.h"
 
-struct virtual_finger_state {
+struct elan_virtual_finger_state {
     TouchCoordinates prev;
     TouchCoordinates now;
     uint8_t pressure;
@@ -240,7 +240,7 @@ private:
 
     int heldFingers = 0;
     int headPacketsCount = 0;
-    virtual_finger_state virtualFinger[ETP_MAX_FINGERS] {};
+    elan_virtual_finger_state virtualFinger[ETP_MAX_FINGERS] {};
 
     static_assert(ETP_MAX_FINGERS <= kMT2FingerTypeLittleFinger, "Too many fingers for one hand");
 

--- a/VoodooPS2Trackpad/VoodooPS2SynapticsTouchPad.cpp
+++ b/VoodooPS2Trackpad/VoodooPS2SynapticsTouchPad.cpp
@@ -807,7 +807,7 @@ void ApplePS2SynapticsTouchPad::assignVirtualFinger(int physicalFinger) {
         return;
     }
 	for (int j = 0; j < SYNAPTICS_MAX_FINGERS; j++) {
-		virtual_finger_state &vfj = virtualFingerStates[j];
+		synaptics_virtual_finger_state &vfj = virtualFingerStates[j];
         if (!vfj.touch) {
             fingerStates[physicalFinger].virtualFingerIndex = j;
             vfj.touch = true;
@@ -819,7 +819,7 @@ void ApplePS2SynapticsTouchPad::assignVirtualFinger(int physicalFinger) {
 	}
 }
 
-void ApplePS2SynapticsTouchPad::assignFingerType(virtual_finger_state &vf) {
+void ApplePS2SynapticsTouchPad::assignFingerType(synaptics_virtual_finger_state &vf) {
 	vf.fingerType = kMT2FingerTypeUndefined;
 	for (MT2FingerType i = kMT2FingerTypeIndexFinger; i < kMT2FingerTypeCount; i = (MT2FingerType)(i + 1))
 		if (freeFingerTypes[i]) {
@@ -1511,7 +1511,7 @@ bool ApplePS2SynapticsTouchPad::renumberFingers() {
             IOLog("synaptics_parse_hw_state: ERROR: invalid physical finger %d", fi.virtualFingerIndex);
             continue;
         }
-        virtual_finger_state &fiv = virtualFingerStates[fi.virtualFingerIndex];
+        synaptics_virtual_finger_state &fiv = virtualFingerStates[fi.virtualFingerIndex];
         fiv.x_avg.filter(fi.x);
         fiv.y_avg.filter(fi.y);
         fiv.width = fi.w;

--- a/VoodooPS2Trackpad/VoodooPS2SynapticsTouchPad.h
+++ b/VoodooPS2Trackpad/VoodooPS2SynapticsTouchPad.h
@@ -170,7 +170,7 @@ struct synaptics_hw_state {
  transducers заполняются каждый раз с номером виртуального пальца столько, сколько надо.
  Будут ли при этом отжиматься отпущенные пальцы?
  */
-struct virtual_finger_state {
+struct synaptics_virtual_finger_state {
     SimpleAverage<int, 5> x_avg;
     SimpleAverage<int, 5> y_avg;
     uint8_t pressure;
@@ -239,7 +239,7 @@ private:
     uint32_t physical_max_y {0};
 
     synaptics_hw_state fingerStates[SYNAPTICS_MAX_FINGERS] {};
-    virtual_finger_state virtualFingerStates[SYNAPTICS_MAX_FINGERS] {};
+    synaptics_virtual_finger_state virtualFingerStates[SYNAPTICS_MAX_FINGERS] {};
     bool freeFingerTypes[kMT2FingerTypeCount];
 
     bool disableDeepSleep {false};
@@ -247,7 +247,7 @@ private:
     static_assert(SYNAPTICS_MAX_FINGERS <= kMT2FingerTypeLittleFinger, "Too many fingers for one hand");
 
     void assignVirtualFinger(int physicalFinger);
-    void assignFingerType(virtual_finger_state &vf);
+    void assignFingerType(synaptics_virtual_finger_state &vf);
     int lastFingerCount;
     int lastSentFingerCount;
     bool hadLiftFinger;

